src/dds/rs-dds-internal-data.cpp

```diff
@@ -0,0 +1,178 @@
+    else if( product_id == RS457_PID )
+    {
+        tags.push_back( { RS2_STREAM_COLOR, -1, 640, 480, RS2_FORMAT_RGB8, 30, profile_tag::PROFILE_TAG_SUPERSET | profile_tag::PROFILE_TAG_DEFAULT } );
+        tags.push_back( { RS2_STREAM_DEPTH, -1, 640, 480, RS2_FORMAT_Z16, 30, profile_tag::PROFILE_TAG_SUPERSET | profile_tag::PROFILE_TAG_DEFAULT } );
+    }
+    else if( product_line == "D400" )
+    {
+        tags.push_back( { RS2_STREAM_DEPTH, -1, 1280, 720, RS2_FORMAT_Z16, 30, profile_tag::PROFILE_TAG_SUPERSET | profile_tag::PROFILE_TAG_DEFAULT } );
+        tags.push_back( { RS2_STREAM_INFRARED, 1, 1280, 720, RS2_FORMAT_RGB8, 30, profile_tag::PROFILE_TAG_SUPERSET | profile_tag::PROFILE_TAG_DEFAULT } );
+        tags.push_back( { RS2_STREAM_INFRARED, 2, 1280, 720, RS2_FORMAT_RGB8, 30, profile_tag::PROFILE_TAG_SUPERSET } );
+    }
+
+    // For other devices empty tags will be returned, no default profile defined.
+
+    return tags;
+}
+
+
+std::vector< processing_block_factory > dds_rs_internal_data::get_profile_converters( const std::string & product_id,
+                                                                                      const std::string & product_line )
+{
+    std::vector< processing_block_factory > factories;
+    std::vector< processing_block_factory > tmp;
+
+    if( product_line == "D400" )
+    {
+        if( product_id == RS457_PID )
+        {
+            tmp = processing_block_factory::create_pbf_vector< uyvy_converter >( RS2_FORMAT_UYVY,
+                                                                                 target_formats( RS2_FORMAT_UYVY ),
+                                                                                 RS2_STREAM_COLOR );
+            for( auto & it : tmp )
+                factories.push_back( std::move( it ) );
+            tmp = processing_block_factory::create_pbf_vector< uyvy_converter >( RS2_FORMAT_YUYV,
+                                                                                 target_formats( RS2_FORMAT_YUYV ),
+                                                                                 RS2_STREAM_COLOR );
+            for( auto & it : tmp )
+                factories.push_back( std::move( it ) );
+            factories.push_back( { { { RS2_FORMAT_Y12I } },
+                                   { { RS2_FORMAT_Y16, RS2_STREAM_INFRARED, 1 },
+                                     { RS2_FORMAT_Y16, RS2_STREAM_INFRARED, 2 } },
+                                   []() { return std::make_shared< y12i_to_y16y16_mipi >(); } } );
+        }
+        else
+        {
+            tmp = processing_block_factory::create_pbf_vector< yuy2_converter >( RS2_FORMAT_YUYV,
+                                                                                 target_formats( RS2_FORMAT_YUYV ),
+                                                                                 RS2_STREAM_COLOR );
+            for( auto & it : tmp )
+                factories.push_back( std::move( it ) );
```

Comment:
Why not create directly in `factories`?

Reply:
Since `create_pbf_vector` returns more than one object to insert into `factories` I cannot use `emplace`. I think that moving using `push_back` is better than copying using `insert`

Reply:
I meant 3 lines above, just assign to `factories` instead of manually moving item by item.